### ISSUE TITLE
Relevant lines incorrect for unloaded files?

### ISSUE THREAD

**Issue by odlp** [colszowka/simplecov#604]
Working on a legacy codebase with no-previous tests, I noticed that unloaded files don't have relevant lines classified correctly. Whitespace and comments are classified as relevant, thereby overstating the total number of uncovered lines for the codebase.

I've made a repo to [reproduce the issue](https://github.com/odlp/simplecov-bug).

### Actual result:

 ![](https://github.com/odlp/simplecov-bug/raw/master/actual.png)

### Expected result:

![](https://github.com/odlp/simplecov-bug/raw/master/expected.png)

## Source of issue

I believe this issue is caused by [`SimpleCov#add_not_loaded_files`](https://github.com/colszowka/simplecov/blob/c615cd6778df799e51840538d150ef1621deb77b/lib/simplecov.rb#L60:L71):

```ruby
def add_not_loaded_files(result)
  if tracked_files
    result = result.dup
    Dir[tracked_files].each do |file|
      absolute = File.expand_path(file)

      result[absolute] ||= [0] * File.foreach(absolute).count
    end
  end

  result
end
```

This always creates an array of zeros, but for whitespace / comment lines this should be `nil` to satisfy the downstream [`SourceFile::Line`](https://github.com/colszowka/simplecov/blob/master/lib/simplecov/source_file.rb#L9) code:

> the coverage being either nil (coverage not applicable, e.g. comment line), 0 (line not covered) or >1 (the amount of times the line was executed

I'm happy to make a PR with a fix for this but would appreciate confirmation this issue is valid.

**Comment by odlp** [comment]
Just realised this is probably related to #600

**Comment by PragTob** [comment]
Closing in favor of #600 and fix should be incoming with #605 :)
